packages/concerto-cto/lib/printer.js

```diff
@@ -197,28 +197,28 @@ function modifiersFromMetaModel(mm){
 }
 
 /**
- * Create a property string from a metamodel
- * @param {object} mm - the metamodel
- * @return {string} the string for that property
+ * Create a property string from a metamodel property
+ * @param {object} prop - the property in scope
+ * @return {string} the CML string representation of the property
  */
-function propertyFromMetaModel(mm) {
+function propertyFromMetaModel(prop) {
     let result = '';
 
-    if (mm.decorators) {
-        result += decoratorsFromMetaModel(mm.decorators, '  ');
+    if (prop.decorators) {
+        result += decoratorsFromMetaModel(prop.decorators, '  ');
     }
-    if (mm.$class === `${MetaModelNamespace}.RelationshipProperty`) {
+    if (prop.$class === `${MetaModelNamespace}.RelationshipProperty` || prop.isRelationship) {
```

Comment:
This suggests to me that the metamodel is wrong.

Did you consider boxing a relationship to a Map inside a RelationshipProperty?

As it is, I believe that this will give us two ways to represent relationships to things.
1. Using a `RelationshipProperty`
2. Using a `MapDeclaration` with `isRelationship` set to `true`

Reply:
On rereading your comment it sounds like you mean something like:

```
concept RelationshipProperty extends Property {
  o TypeIdentifier type
  o  MapValue 
}
```

which still inherits the design issues pointed out above.

I'm curious why you think the initial solution may be wrong based on the branch

`    if (prop.$class === `${MetaModelNamespace}.RelationshipProperty` || prop.isRelationship) {
`
Why does this suggest the metamodel is incorrect?

Reply:
@mttrbrts changes added based on prior discussion. You can find the updated metamodel for reference [here](https://github.com/accordproject/concerto-metamodel/pull/2)